src/command_type.h

```diff
@@ -190,6 +190,7 @@ enum Commands : uint16 {
 	CMD_BUILD_SINGLE_RAIL,            ///< build a single rail track
 	CMD_REMOVE_SINGLE_RAIL,           ///< remove a single rail track
 	CMD_LANDSCAPE_CLEAR,              ///< demolish a tile
+	CMD_LANDSCAPE_MAGIC_CLEAR,        ///< demolish a tile with magic bulldozer behaviour
```

Comment:
As magic bulldozer is a command flag set by a scope mode, I don't think adding a new command should be necessary.

Reply:
Probably needs handling of mode in `ScriptObject::DoCommandPrep()` and a way to add the flag during execution (I don't think any script call ever touched the flags)

Reply:
Presumably Test/Exec scope mode also sets the appropriate flags for that, so it should be similar to that?

Reply:
Test/Exec just set the `estimate_only` boolean, which is then passed through `Command::Unsafe()` and `Command::Execute()`. `DC_EXEC` handling is done in `Command::Execute()`.

I used `Command::` to simplify, it's actually the templated command stuff.